### PR TITLE
QPACK SC:  No encoder memory for blocked

### PR DESCRIPTION
**Pull request by MikeBishop**
Per discussion on Slack, expanding this discussion in the Security Considerations to be more accurate.  Editorial, since these are purely statements about the properties of the design.

### CODE REVIEW

**Comment by dtikhonov** [comment]
> An encoder allocates memory to track all dynamic table references in unacknowledged representations.  Implementations can directly limit the amount of state memory by only using as many references to the dynamic table as it wishes to track; no signaling to the decoder is required.

This is true, but this is also obvious, at least once you start implementing the encoder.  Too, there is the natural limit on dynamic references: it is the function of the product of the number of allowed streams and the number of entries in the dynamic table (which itself is a function of the table size).

I am concerned that this new verbiage puts the wrong thought into the head of the implementer; we should not encourage this.  Limiting the number of dynamic table references (done out of memory conservation zeal) affects compression performance directly -- it worsens it.

Perhaps we can tack on the following sentence to the end of the paragraph:

> Note that limiting the number of references is likely to affect performance negatively.

**Comment by martinthomson** [comment]
I'm OK with that, though I would qualify with "compression performance", as performance has other dimensions that the extra state might impinge upon.

**Comment by dtikhonov** [comment]
Another thing to note is that sometimes dynamic table references do not have to have their own objects in the encoder.  For examples, they are [virtually free](https://github.com/litespeedtech/ls-qpack/blob/v2.2.0/lsqpack.c#L255L256) in ls-qpack.

(Everyone reading this in 2022: you're welcome)

**Comment by dtikhonov** [comment]
Yeah, compression performance qualification is a good suggestion.

File: draft-ietf-quic-qpack.md [hunk 1]
```diff
@@ -1362,9 +1362,14 @@ allows (see {{eviction}}).
...
 setting an appropriate value for the maximum number of blocked streams.  In
 HTTP/3, this is realized by setting an appropriate value for the
-QPACK_BLOCKED_STREAMS parameter.  An encoder can limit the amount of state
-memory by only using as many blocked streams as it wishes to support; no
-signaling to the decoder is required.
+QPACK_BLOCKED_STREAMS parameter.  Streams which risk becoming blocked consume no
+additional state memory on the encoder.
+
+An encoder allocates memory to track all dynamic table references in
+unacknowledged representations.  Implementations can directly limit the amount
```
Comment (afrind) [inline draft-ietf-quic-qpack.md:1369]: representations are not acknowledged, encoded field section blocks are, right?

File: draft-ietf-quic-qpack.md [hunk 2]
```diff
@@ -1362,9 +1362,14 @@ allows (see {{eviction}}).
...
+of state memory by only using as many references to the dynamic table as it
```
Comment (afrind) [inline draft-ietf-quic-qpack.md:1370]: Subject/pronoun agreement: "Implementations can...as it wishes".  Perhaps "An implementation can..."

**Pull request merged**